perfschema/instrument.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perfschema
+
+import (
+	"sync/atomic"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+// EnumCallerName is used as a parameter to avoid calling runtime.Caller(1) since
+// it is too expensive (500ns+ per call), we don't want to invoke it repeatedly for
+// each instrument.
+type EnumCallerName int
+
+const (
+	// CallerNameSessionExecute is for session.go:Execute() method.
+	CallerNameSessionExecute EnumCallerName = iota
+)
+
+const (
+	stageInstrumentPrefix       = "stage/"
+	statementInstrumentPrefix   = "statement/"
+	transactionInstrumentPrefix = "transaction"
```

Comment:
Same as MySQL, so far I am not sure why MySQL's transaction instruments does not have.